File: lingting-core/src/main/java/live/lingting/framework/api/ScrollParams.java
package live.lingting.framework.api;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

/**
 * @author lingting 2024-02-02 17:54
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
public class ScrollParams<T> {

	private long size = 10;

	private T cursor;

	public long getSize() {
		return size < 1 ? 10 : size;
	}

}

File: lingting-core/src/main/kotlin/live/lingting/framework/api/ScrollParams.kt
package live.lingting.framework.api

/**
 * @author lingting 2024-02-02 17:54
 */
data class ScrollParams<T>(
    var size: Long,
    var cursor: T? = null,
) {

    constructor() : this(10)

    constructor(size: Long) : this(size, null)

}
